packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -445,4 +445,27 @@ export class CacheService {
     }
     return values.slice(start, end + 1);
   }
+
+  /**
+   * Retrieves all keys matching the given pattern.
+   * @param {string} pattern - The pattern to match keys against.
+   * @param {string} callingMethod - The name of the calling method.
+   * @param {string} [requestIdPrefix] - A prefix to include in log messages (optional).
+   * @returns {Promise<string[]>} A Promise that resolves with an array of keys that match the pattern.
+   */
+  async keys(pattern: string, callingMethod: string, requestIdPrefix?: string): Promise<string[]> {
+    if (this.isSharedCacheEnabled && this.sharedCache instanceof RedisCache) {
+      try {
+        return await this.sharedCache.keys(pattern, callingMethod, requestIdPrefix);
+      } catch (error) {
+        const redisError = new RedisCacheError(error);
+        this.logger.error(
+          `${requestIdPrefix} Error occurred while getting keys from Redis. Error is: ${redisError.fullError}`,
+        );
```

Comment:
We can add the `redisError` as the first argument of the error log so it can show full error.
```suggestion
        this.logger.error(
          redisError,
          `${requestIdPrefix} Error occurred while getting keys from Redis.`,
        );
```

Reply:
This wouldn't be consistent with the logging style for all other methods in this class

Reply:
Ah, yes, I noticed that. I suggest applying the new style to the error logging in this class. Instead of embedding and converting `redisError` to a string, the new style would directly pass the error object, allowing the logger to capture the full error context. This approach is more informative, as the logger can access all properties of the error object.

Reply:
Okay, I will change all logs to this style then 😁

Reply:
Done. ✅